lib/web_of_science/process_records.rb

```diff
@@ -34,42 +34,37 @@ def execute
 
       # @return [Array<String>] WosUIDs that create a new Publication
       def create_publications
-        new_records = select_new_wos_records(filter_databases) # cf. WebOfScienceSourceRecord
-        new_records = save_wos_records(new_records) # save WebOfScienceSourceRecord
-        new_records = filter_by_identifiers(new_records) # cf. PublicationIdentifier
-        new_records = new_records.select { |rec| create_publication(rec) }
-        pubmed_additions(new_records)
-        new_records.map(&:uid)
+        filter_databases # cf. Settings.WOS.ACCEPTED_DBS
+        select_new_wos_records # cf. WebOfScienceSourceRecord
+        save_wos_records # save WebOfScienceSourceRecord
+        filter_by_identifiers # cf. PublicationIdentifier
+        records.select! { |rec| create_publication(rec) }
+        pubmed_additions
+        records.map(&:uid)
       end
 
       ## 1
       # Filter and select new WebOfScienceSourceRecords
-      # @return [Array<WebOfScience::Record>]
       def filter_databases
-        return [] if records.count.zero?
-        return records if Settings.WOS.ACCEPTED_DBS.empty?
-        records.select { |rec| Settings.WOS.ACCEPTED_DBS.include? rec.database }
+        return if records.empty? || Settings.WOS.ACCEPTED_DBS.empty?
```

Comment:
This `Settings.WOS.ACCEPTED_DBS.empty?` check should be moved up into `#execute`.  No point even connecting if we aren't going to accept any records returned!

Reply:
Ah, yea; this is probably a cause to raise an exception actually, smells of some kind of oversight in creating or changing the settings file.

Reply:
Pushed up an exception the `Settings.WOS.ACCEPTED_DBS.empty?`